src/Umbraco.Core/Services/Implement/FileService.cs

```diff
@@ -405,6 +405,21 @@ public long GetScriptFileSize(string filepath)
         /// <returns></returns>
         public ITemplate CreateTemplateWithIdentity(string name, string alias, string content, ITemplate masterTemplate = null, int userId = Constants.Security.SuperUserId)
         {
+            if(name == null)
+            {
+                throw new ArgumentNullException(nameof(name), "Name cannot be null or empty");
```

Comment:
It's better to use the default message 'Value cannot be null.' here:

```suggestion
                throw new ArgumentNullException(nameof(name));
```

Reply:
Just to clarify: not passing a message will use the aforementioned default message 👍

Reply:
Aah! Missed my eyes :(